### PR TITLE
Added many micro frameworks from Jodd suite.

### PR DESCRIPTION
**Pull request by igorspasic**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,6 +119,7 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 
 *Libraries related to handling date and time.*
 
+* [JDateTime](http://jodd.org/doc/jdatetime.html) - Developer-friendly, astronomically precise date/time class based on Julian dates.
```
Comment (akullpp) [inline README.md]: Why use JDateTime over Java 8 or Joda-Time?

Comment (igorspasic) [inline README.md]: + only few classes in the package
+ it also gives you [Julian date number](http://en.wikipedia.org/wiki/Julian_day) (JDN), so it is very easy to calc distance between two time points - just using add/sub. Distance between two JDN is _always_ 1.
+ Uses astronomical algorithm; hence very stable and precise

btw, it exist even before JodaTime :)

**Comment by akullpp** [comment]
I can't add a dozen libraries posted by the author. Either other people comment on this or you could add Jodd in its entirety as a single entry.

**Comment by igorspasic** [comment]
I don't understand, please explain the reason. I don't see what is the problem.

Jodd is a _bundle_ of tools and libraries. It is NOT a single library. It is like you have spring, hibernate, struts, jaspect, json, commons, http client.... under the same name. They can be used separately, or together.

Each of added library is complete and functional; and I didn't list those that are still not battle-tested. There is absolutely no difference if I added 10 different libraries or 10 libraries that exist under the same umbrella, like Jodd.

Moreover, there is no reasonable category where just 'Jodd' could fit in as a single entry.

**Comment by akullpp** [comment]
I understand, but nevertheless I'd like other opinions too, let's keep this open for a few days and then add them one by one.

Btw, I wouldn't add every Spring library as well.

File: README.md (hunk 1 above)
Comment (akullpp) [inline README.md]: I really can't see the reason to prefer JDateTime over Joda-Time or Java 8. Joda-Time has a convenient API which encapsulates the complexity and provides a rather simple interface. It also provides the featue of the edge case of Julian dates.

Comment (igorspasic) [inline README.md]: no problem, this PR is just suggestion after all :)

File: README.md [hunk 2]
```diff
@@ -404,16 +425,19 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
...
 ## Web Frameworks
 
 *Frameworks that handle the communication between the layers of an web application.*
-h
+
 * [Apache Tapestry](http://tapestry.apache.org/) - Component oriented framework for creating dynamic, robust, highly scalable web applications in Java.
 * [Apache Wicket](http://wicket.apache.org/) - Component-based web application framework similar to Tapestry with a stateful GUI.
 * [Google Web Toolkit](http://www.gwtproject.org/) - Toolbox which includes a Java-to-JavaScript compiler for client-side code, XML parser, API for RPC, JUnit integration, internationalization support and widgets for the GUI.
 * [Grails](https://grails.org/) - Groovy framework with the aim to provide a highly productive environment by favoring convention over configuration, no XML and support for mixins.
+* [Madvoc](http://jodd.org/doc/madvoc/index.html) - Fully-featured micro MVC web framework.
```
Comment (akullpp) [inline README.md]: Why should we add another web framework. I really don't understand how it is different and at what it excels.

File: README.md [hunk 3]
```diff
@@ -404,16 +425,19 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 
 * [Apache Nutch](http://nutch.apache.org/) - Highly extensible, highly scalable Web crawler for production environment.
 * [Crawler4j](https://code.google.com/p/crawler4j/) - Simple lightweight alternative.
+* [Jerry](http://jodd.org/doc/jerry/index.html) - jQuery in Java.
 * [JSoup](http://jsoup.org/) - Scrapes, parses, manipulates and cleans HTML.
+* [Lagarto](http://jodd.org/doc/lagarto/index.html) - All purpose (X)HTML parser.
```
Comment (akullpp) [inline README.md]: Why another one?

File: README.md [hunk 4]
```diff
@@ -394,6 +414,7 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 
 * [args4j](http://args4j.kohsuke.org/) - Command line arguments parser
 * [Apache Commons](http://commons.apache.org/) - Provides different general purpose functions like configuration, validation, collections, file upload or XML processing.
+* [BeanUtil](http://jodd.org/doc/beanutil.html) - Rich, bean utility library, with types conversion.
```
Comment (akullpp) [inline README.md]: Wow that name is really confusing since there is also BeanUtils

File: README.md [hunk 5]
```diff
@@ -286,6 +305,7 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 *Libraries for network programming.*
 
 * [Grizzly](https://grizzly.java.net/) - NIO framework. Used as a network layer in Glassfish.
+* [Jodd HTTP](http://jodd.org/doc/http.html) - developer-friendly fluent HTTP library.
```
Comment (akullpp) [inline README.md]: The naming is really bad. HTTP, EMail, JSON. Only discernible by identifying it at the same time with Jodd.

File: README.md [hunk 6]
```diff
@@ -248,6 +261,12 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 * [H2O](http://0xdata.com/) - Analytics engine for statistics over big data.
 * [Weka](http://www.cs.waikato.ac.nz/ml/weka/) - Collection of algorithms for data mining tasks ranging from pre-processing to visualization.
 
+## Mail
+
+*Libraries that simplify sending and receiving emails.*
+
+* [Jodd EMail](http://jodd.org/doc/email.html) - Java mail, fluent and simplified.
```
Comment (akullpp) [inline README.md]: I don't understand the benefit as compared to the simple mail client of Apache Commons.

File: README.md [hunk 7]
```diff
@@ -219,6 +231,7 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 
 * [Gson](https://code.google.com/p/google-gson/) - Serializes Java objects to JSON and vice versa. Good performance with on-the-fly usage.
 * [Jackson](http://wiki.fasterxml.com/JacksonHome) - Similar to GSON but has performance gains if you need to instantiate the library more often.
+* [Jodd JSON](http://jodd.org/doc/json/index.html) - Easy-to-use JSON de/serializer with many features.
```
Comment (akullpp) [inline README.md]: Why? Comparison in regards to other libraries, please.

File: README.md [hunk 8]
```diff
@@ -118,6 +129,7 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 
 * [Dagger](http://square.github.io/dagger/) - Compile-time injection framework without reflection, mainly for Android.
 * [Guice](https://github.com/google/guice) - Lightweight but powerful framework that completes Dagger.
+* [Petite](http://jodd.org/doc/petite/index.html) - micro DI container.
```
Comment (akullpp) [inline README.md]: Uppercase at sentence initial letter please.

File: README.md [hunk 9]
```diff
@@ -63,6 +65,7 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 
 * [ASM](http://asm.ow2.org/) - All purpose, low level, bytecode manipulation and analysis.
 * [Byte Buddy](http://bytebuddy.net/) - Further simplifies bytecode generation with a fluent API.
+* [Proxetta](http://jodd.org/doc/proxetta/index.html) - Simple dynamic proxies, unique in the way how pointcuts and advices are defined.
```
Comment (akullpp) [inline README.md]: I like it, you described how it is different.

File: README.md [hunk 10]
```diff
@@ -81,6 +84,12 @@ A curated list of awesome Java frameworks, libraries and software. Inspired by o
 * [ANTLR](http://www.antlr.org/) - Complex full-featured framework for top-down parsing.
 * [JavaCC](https://javacc.java.net/) - More specific and slightly easier to learn. Has syntactic lookahead.
 
+## Configuration
+
+*Libraries that provide external configuration.*
+
+* [Props](http://jodd.org/doc/props.html) - Properties on steroids.
```
Comment (akullpp) [inline README.md]: Also an interesting library and we could use the section.

**Comment by akullpp** [comment]
Overall I like your framework but you will need to point out the advantages compared to other libraries more and work on the naming of the modules.

**Comment by akullpp** [comment]
So I would definitely add Proxeta and Props, the other libraries are discussable but for some I just can't find a reason, e.g your date lib.

**Comment by igorspasic** [comment]
Sure, no problem.

**Comment by akullpp** [comment]
Will add the tools as soon as I have time or you do another PR.

**Pull request closed without merging**